test/vanilla/legacy/AcceptanceTests/test_xms_error.py

```diff
@@ -98,6 +98,39 @@ def test_error_deserialization_with_param_name_models(self, client):
         assert isinstance(excinfo.value.model, PetSadError)
         assert excinfo.value.status_code == 500
 
+    def test_failsafe_deserialize(self, client):
```

Comment:
added test here. Didn't add in `msrest` bc it seems that `msrest` tests deliberately don't include `azure-core`. So thought it best to add here instead